src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php

```diff
@@ -96,17 +169,47 @@ private function updateOrderInvoice(OrderDetail $orderDetail)
     /**
      * @param Order $order
      * @param OrderDetail $orderDetail
+     * @param Cart $cart
      *
      * @return bool
      */
-    private function updateOrder(Order $order, OrderDetail $orderDetail)
+    private function updateOrder(Order $order, OrderDetail $orderDetail, Cart $cart)
     {
         // @todo: use https://github.com/PrestaShop/decimal for price computations
-        $order->total_paid -= $orderDetail->total_price_tax_incl;
-        $order->total_paid_tax_incl -= $orderDetail->total_price_tax_incl;
-        $order->total_paid_tax_excl -= $orderDetail->total_price_tax_excl;
-        $order->total_products -= $orderDetail->total_price_tax_excl;
-        $order->total_products_wt -= $orderDetail->total_price_tax_incl;
+        $computingPrecision = new ComputingPrecision();
```

Comment:
I think you copied this code from the `AddProductToOrderHandler` right?
Then maybe it would be worth having a dedicated service/class to perform those updates? Some kind `OrderProductUpdater` which would update the cart at the same time (allowing you to store cart rules before/after update in a class field).

Reply:
A new issue has been created : https://github.com/PrestaShop/PrestaShop/issues/18432